processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/FragmentCompletionTimeDeriver.java

```diff
@@ -90,4 +95,16 @@ public CompletionTime processOneToOne(String tenantId, Span item) throws RetryAt
         return null;
     }
 
+    private void setFault(CompletionTime completionTime, List<BinaryAnnotation> binaryAnnotations) {
+        List<HttpCodesUtil.HttpCode> errorCodes =
+                HttpCodesUtil.getClientOrServerErrors(HttpCodesUtil.getHttpStatusCodes(binaryAnnotations));
+
+        if (errorCodes.size() > 0) {
```

Comment:
It's still not clear if you expect only one error code (and thus, the method should return a single object instead of a list), or if you expect a list of error codes. If a list is expected, you could join the messages.